src/utilities/oas-schema/oas-schema.ts

```diff
@@ -1,35 +1,123 @@
-import swaggerClient, { Response, Swagger } from 'swagger-client';
+import swaggerClient, { Request, Response, Swagger } from 'swagger-client';
 import OASOperation, { OASOperationFactory } from '../oas-operation';
 import ExampleGroup from '../example-group';
+import {
+  OASSecurityScheme,
+  OASSecurityFactory,
+  OASSecurity,
+  OASSecurityType,
+  OASIn,
+} from '../oas-security';
 
 class OASSchema {
   private _client: Promise<Swagger>;
 
+  private _swaggerOptions: Parameters<typeof swaggerClient>[0];
+
   private operations: OASOperation[];
 
+  private topSecurities: OASSecurity[];
+
+  private securitySchemes: OASSecurityScheme[];
+
   constructor(options: Parameters<typeof swaggerClient>[0]) {
     this._client = swaggerClient(options);
+    this._swaggerOptions = options;
     this.operations = [];
+    this.topSecurities = [];
+    this.securitySchemes = [];
   }
 
   public set client(client: Promise<Swagger>) {
     this._client = client;
   }
 
+  public get client(): Promise<Swagger> {
+    return this._client;
+  }
+
   execute = async (
     operation: OASOperation,
     exampleGroup: ExampleGroup,
+    flags?: { apiKey: string | undefined },
   ): Promise<Response> => {
+    const requestInterceptor = await this.getOperationRequestInterceptor(
+      operation,
+      flags,
+    );
     const schema = await this._client;
-
-    return schema
+    const response = schema
       .execute({
         operationId: operation.operationId,
         parameters: exampleGroup.examples,
+        requestInterceptor,
```

Comment:
if we handle all the security logic at the operation level we should be able to remove this interceptor. We could just do
```ts
.execute({
    operationId: operation.operationId,
    parameters: exampleGroup.examples,
    securities: operation.securities.reduce((securities, security) => {
        return {
            ....securities,
            security.key: this.securityFor(security), // we'd have to write this method, but it seems easier to maintain than the interceptor
        }
    }, {})
})
```